events/filters.py

```diff
@@ -508,6 +538,86 @@ def filter_registration_admin_user(self, queryset, name, value):
         else:
             return queryset.none()
 
+    def filter_days(self, queryset, name, value):
+        try:
+            days = int(value)
+        except ValueError:
+            raise ParseError(_("Error while parsing days."))
+        if days < 1:
+            raise serializers.ValidationError(_("Days must be 1 or more."))
+        if "start" in self.data or "end" in self.data:
+            raise serializers.ValidationError(
+                _("Start or end cannot be used with days.")
+            )
+
+        today = datetime.now(datetime_timezone.utc).date()
+
+        start = today.isoformat()
+        end = (today + timedelta(days=days)).isoformat()
+
+        queryset = self.filter_start(queryset, "start", start)
+        queryset = self.filter_end(queryset, "end", end)
+        return queryset
+
+    def filter_start(self, queryset, name, value):
+        if "end" not in self.data:
+            # postponed events are considered to be "far" in the future and should be
+            # included if end is *not* given
+            postponed_q = Q(event_status=Event.Status.POSTPONED)
+        else:
+            postponed_q = Q()
+
+        # Inconsistent behaviour, see test_inconsistent_tz_default
+        dt = utils.parse_time(value, default_tz=pytz.timezone(settings.TIME_ZONE))[0]
+
+        # only return events with specified end times, or unspecified start times, during the whole of the event  # noqa: E501
+        # this gets of rid pesky one-day events with no known end time (but known
+        # start) after they started
+        return queryset.filter(
+            Q(end_time__gt=dt, has_end_time=True)
+            | Q(end_time__gt=dt, has_start_time=False)
+            | Q(start_time__gte=dt)
+            | postponed_q
+        )
+
+    def filter_end(self, queryset, name, value):
+        dt = utils.parse_end_time(value, default_tz=pytz.timezone(settings.TIME_ZONE))[
+            0
+        ]
+        return queryset.filter(Q(end_time__lt=dt) | Q(start_time__lte=dt))
+
+    def filter_hide_recurring_children(self, queryset, name, value: Optional[bool]):
```

Comment:
I'd like to see a docstring here reminding that this is a special filter that is very order dependent on how it affects the result.

Reply:
somekind of string of doc has been inserted.